Tella/Data/Networking/APICall/APIError.swift

```diff
@@ -11,35 +11,61 @@ enum APIError: Swift.Error {
     case unexpectedResponse
     case noInternetConnection
     case badServer
+    case noToken
+    case driveApiError(Error)
 }
 
 extension APIError: LocalizedError {
 
     var errorDescription: String? {
         switch self {
         case .invalidURL:
-            return "Invalid URL"
+            return LocalizableError.invalidUrl.localized
         case let .httpCode(code):
             return customErrorMessage(errorCode: code)
         case .unexpectedResponse:
-            return "Unexpected response from the server"
+            return LocalizableError.unexpectedResponse.localized
         case .noInternetConnection:
             return LocalizableSettings.settServerNoInternetConnection.localized
         case .badServer:
             return LocalizableSettings.settServerServerURLIncorrect.localized
+        case .noToken:
+            return LocalizableSettings.settServerNoTokenPresent.localized
+        case .driveApiError(let error):
+            return customDriveErrorMessage(error: error)
         }
     }
     private func customErrorMessage(errorCode : Int) -> String {
         let httpErrorCode = HTTPErrorCodes(rawValue: errorCode)
         switch httpErrorCode{
         case .unauthorized:
-            return "Invalid username or password"
+            return LocalizableError.unauthorized.localized
         case .forbidden:
-            return "Account locked due to too many unsuccessful attempts."
+            return LocalizableError.forbidden.localized
         case .notFound:
             return LocalizableSettings.settServerServerURLIncorrect.localized
         default:
-            return "Unexpected response from the server"
+            return LocalizableError.unexpectedResponse.localized
         }
     }
+    
+    private func customDriveErrorMessage(error: Error) -> String {
+        if let nsError = error as NSError? {
+            switch nsError.domain {
+            case GoogleAuthConstants.GTLRErrorObjectDomain:
+                let errorCode = nsError.code
+                let errorMessage = nsError.localizedDescription
+                            
+                return customErrorMessage(errorCode: errorCode)
+            case GoogleAuthConstants.HTTPStatus:
+                return customErrorMessage(errorCode: nsError.code)
```

Comment:
I think it's better to add the custom Google Drive error messages since the purpose of adding `customDriveErrorMessage` is to display Google Drive's own messages.

Reply:
got it, I was planning to improve this in another iteration with more time. I add the main basic localizable errors which are for authorization and permission. If there is other error we will show the localized description of gDrive, those are a little bit technical sometimes but we can improve those later when we have more bandwith